### ISSUE TITLE
New sample: Kotlin version of todo-mvp

### ISSUE THREAD

**Issue by aballano** [googlesamples/android-architecture#235]
Although is not an architectural pattern, I think it could be nice to have a Kotlin version for some of the variants. Interesting points would be the amount of code and size of the app. 

I personally started the todo-mvp-kotlin in my forked repo as an example. so far I converted:
* addedittask
* data

Please note that this is still in progress :) and feel free to contribute to it if you're interested!

https://github.com/Shyish/android-architecture/tree/dev-todo-mvp-kotlin

**Comment by JoseAlcerreca** [comment]
Looking forward to comparing APK size, LOC and method count. Please make sure to mention in the README that Kotlin is not officially supported.

**Comment by Shyish** [comment]
So, first version is done, although there could be some more optimizations to be made.

https://github.com/Shyish/android-architecture/tree/dev-todo-mvp-kotlin

@JoseAlcerreca @malmstein Please take a look when you can and let me know what else do you need :)

**Comment by LordRaydenMK** [comment]
Hey @Shyish I was also curious how the app would look written in Kotlin and I started: https://github.com/LordRaydenMK/android-architecture/tree/todo-mvp-kotlin

Looks like you are way ahead of me. I am almost done with the data package. 

I noticed some areas in the code on your fork that could be improved. I would like to discuss them with you and maybe contribute with a pull request. Let me know what you think.
